### PR TITLE
document dependencies

### PR DESCRIPTION
**Pull request by HagarMeir**
Signed-off-by: Hagar Meir <hagar.meir@ibm.com>

### CODE REVIEW

**Comment by HagarMeir** [comment]
https://github.com/SmartBFT-Go/consensus/issues/300

File: pkg/api/dependencies.go [hunk 1]
```diff
@@ -10,45 +10,73 @@ import (
...
 	SendTransaction(targetID uint64, request []byte)
+	// Nodes returns a set of ids of participating nodes.
 	Nodes() []uint64
 }
 
+// Assembler creates proposals.
 type Assembler interface {
+	// AssembleProposal creates a proposal which includes
+	// the given requests (when permitting) and metadata.
 	AssembleProposal(metadata []byte, requests [][]byte) (nextProp bft.Proposal, remainder [][]byte)
```
Comment (C0rWin) [inline pkg/api/dependencies.go:34]: IMO need also elaborate on reminder.

File: pkg/api/dependencies.go [hunk 2]
```diff
@@ -10,45 +10,73 @@ import (
 	protos "github.com/SmartBFT-Go/consensus/smartbftprotos"
 )
 
+// Application delivers the committed proposal and corresponding signatures.
```
Comment (C0rWin) [inline pkg/api/dependencies.go]: Is it indeed committed proposal? I was under impression this is a callback interface which triggers only after replicas have mutual consent.

File: pkg/api/dependencies.go [hunk 3]
```diff
@@ -10,45 +10,73 @@ import (
...
 
+// Verifier validates data and verifies signatures.
 type Verifier interface {
+	// VerifyProposal verifies the given proposal and returns the included requests' info.
 	VerifyProposal(proposal bft.Proposal) ([]bft.RequestInfo, error)
+	// VerifyRequest verifies the given request and returns its info.
 	VerifyRequest(val []byte) (bft.RequestInfo, error)
+	// VerifyConsenterSig verifies the signature for the given proposal.
 	VerifyConsenterSig(signature bft.Signature, prop bft.Proposal) error
+	// 	VerifySignature verifies the signature.
```
Comment (C0rWin) [inline pkg/api/dependencies.go]: A nit, but can you please fix indentation?

File: pkg/api/dependencies.go (hunk 1 above)
Comment (HagarMeir) [reply pkg/api/dependencies.go:34]: Actually the remainder is not used.
https://github.com/SmartBFT-Go/consensus/issues/380
Maybe we should remove it completely.

File: pkg/api/dependencies.go (hunk 2 above)
Comment (HagarMeir) [reply pkg/api/dependencies.go]: This is called after replicas as agreed...after all three phases of consensus. 
Is "committed" not clear enough?

Comment (C0rWin) [reply pkg/api/dependencies.go]: Committed there?

File: pkg/api/dependencies.go (hunk 1 above)
Comment (C0rWin) [reply pkg/api/dependencies.go:34]: If not in use, we definitely need to get rid of it.

File: pkg/api/dependencies.go (hunk 2 above)
Comment (HagarMeir) [reply pkg/api/dependencies.go]: Committed = agreed on
How do you suggest I change it to make it more clear?

Comment (yacovm) [reply pkg/api/dependencies.go]: Consented should be a better term

File: pkg/api/dependencies.go (hunk 1 above)
Comment (yacovm) [reply pkg/api/dependencies.go:34]: :+1:

Comment (HagarMeir) [reply pkg/api/dependencies.go:34]: OK then that will be my next task

**Review by yacovm — APPROVED** [review]

**Pull request merged**